cbqueryx/ensureindexhelper_test.go

```diff
@@ -1,79 +1,122 @@
-package cbqueryx
+package cbqueryx_test
 
 import (
 	"context"
-	"errors"
 	"fmt"
+	"log"
 	"net/http"
 	"testing"
 	"time"
 
-	"github.com/couchbase/gocbcorex/cbmgmtx"
+	"github.com/couchbase/gocbcorex/cbqueryx"
+
 	"github.com/couchbase/gocbcorex/testutils"
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
 )
 
-func TestEnsureQuery(t *testing.T) {
-	testutils.SkipIfShortTest(t)
+func testNodesToNodeTargets(t *testing.T, nodes []testutils.TestNodeTarget) []cbqueryx.NodeTarget {
+	var targets []cbqueryx.NodeTarget
+	for _, node := range nodes {
+		if node.QueryPort == 0 {
+			continue
+		}
+		targets = append(targets, cbqueryx.NodeTarget{
+			Endpoint: fmt.Sprintf("http://%s:%d", node.Hostname, node.QueryPort),
+			Username: testutils.TestOpts.Username,
+			Password: testutils.TestOpts.Password,
+		})
+	}
+	return targets
+}
+
+func TestEnsureQueryDino(t *testing.T) {
+	testutils.SkipIfNoDinoCluster(t)
 
 	ctx := context.Background()
 	transport := http.DefaultTransport
 
-	config, err := cbmgmtx.Management{
-		Transport: transport,
-		UserAgent: "useragent",
-		Endpoint:  "http://" + testutils.TestOpts.HTTPAddrs[0],
-		Username:  testutils.TestOpts.Username,
-		Password:  testutils.TestOpts.Password,
-	}.GetTerseClusterConfig(ctx, &cbmgmtx.GetTerseClusterConfigOptions{
-		OnBehalfOf: nil,
-	})
-	require.NoError(t, err)
-
-	var targets []NodeTarget
-	for _, nodeExt := range config.NodesExt {
-		if nodeExt.Services.N1ql > 0 {
-			targets = append(targets, NodeTarget{
-				Endpoint: fmt.Sprintf("http://%s:%d", nodeExt.Hostname, nodeExt.Services.N1ql),
-				Username: testutils.TestOpts.Username,
-				Password: testutils.TestOpts.Password,
-			})
-		}
-	}
-	if len(targets) == 0 {
-		t.Skip("Skipping test, no query nodes")
+	blockHost, execEndpoint, baseTargets := testutils.SelectNodeToBlock(t, testutils.TestServiceQuery)
+	targets := testNodesToNodeTargets(t, baseTargets)
+
+	if blockHost == "" {
+		// This test requires a very specific cluster setup containing a node with the query service but not indexing.
+		// The test blocks traffic to that query node.
+		t.Skipf("Cluster not in configuration that can be used by this test")
 	}
 
 	// we intentionally use the last target that will be polled as the node
```

Comment:
Same nit

Reply:
This is a required comment, it explains why we skip this test.

Reply:
Nevermind...